### ISSUE TITLE
Plugin / theme lists and multiple networks

### ISSUE THREAD

**Issue by webaware** [bueltge/wordpress-multisite-enhancements#17]
When a multisite has [multiple networks](https://wordpress.org/plugins/wp-multi-network/), the list of sites on which a plugin or theme is active only shows one network -- and it might not be the current network because the list is stored in a site transient.

I propose that either of the following changes be made:

1. show sites from current network where plugin/theme is active, i.e. unique site transient per network
2. show all sites where plugin/theme is active, regardless of network

Option 1 is pretty painless to implement, because it involves adding the site ID to the site transient name. However, I reckon it's more useful to know all sites where a plugin is active, and if one has several networks, that would require visiting each network's plugin admin page to find that out.

Conversely, on a multisite with several networks each hosting many sites, the list could be so big as to be unmanageable. In that case, option 1 might be preferable to option 2.

I'd be happy to submit a PR for either approach, if you can indicate which option you'd prefer, or perhaps suggest something even better.

**Comment by bueltge** [comment]
Thanks for the hint. I think it is helpful, that we have a additional class, there only work, if the multiple network is active. I think it is important, that we not load more on a default wp core. Perhaps it is helpful, that we add hooks, that the new functionality work on the default list style, hook in the column to view plugins, themes for all networks.

**Comment by webaware** [comment]
Splitting off extended functionality for a multi-network install sounds good. My option 2, above, could then be enabled only for those who want it, and other multi-network functionality would have a good home.

That being the case, I reckon my option 1 above should be the default situation, so that the plugin correctly shows which sites within the current network have a plugin/theme activated. The class for managing multiple networks can extend things from that point.

NB: any WP multisite can have multiple networks, whether they use the [WP Multi Network plugin](https://wordpress.org/plugins/wp-multi-network/), [Networks for WordPress](https://wordpress.org/plugins/networks-for-wordpress/), or manually inserting the config into the wp_site table. I think the trigger for offering multiple-network functionality should be > 1 rows in wp_site.

**Comment by bueltge** [comment]
I know only one solution to create multi networks and currently is it also a discussion in the core to realized this. It is possible to check for multi networks? That we not have a active option in the UI to active this. Currently do not have the plugin settings in UIs, is a open wish from different users. But it is a topic for more maintenance effort and support.

**Comment by webaware** [comment]
WP core has basic support for multiple networks already, but no admin UI. Those plugins I linked, above, provide some admin UI and a few helper services. But you can create multiple networks in multisite with just MySQL statements, if you're game. I believe that the WP core discussion is about advancing the native admin UI capabilities and fixing up some cross-network problems.

To check whether a multisite has multiple networks, look at the wp_site table. It will have only one row normally, but will have > 1 rows when there are multiple networks.

I know what you mean about extra maintenance and effort. Perhaps all that is necessary is to implement separate transients for each network so that the list of sites where a plugin/theme is active operates correctly. 

For my own use, I only have two networks on each of the multi-network installs I manage. I'd imagine that someone running a large multi-network multisite would pay for a more advanced tool, or more likely develop something business-specific in-house to suit their needs.

**Comment by bueltge** [comment]
> WP core has basic support for multiple networks already, but no admin UI.
Right. I have read the discussion in slack core, that implement this in the core.

> look at the wp_site table
Fine, thats is the point to different it.

Transients are important for large network. The plugin have this in all jobs, like plugin-, theme usage.

Thanks for your help on the plugin, makes much better.

**Comment by hilfans** [comment]
i just install, but it show on All Inactive Plugin at Usage Column “The site is experiencing technical difficulties. Please check your site admin email inbox for instructions.”

the mail service is not active for my site, so I cant check anyway.

And it make I cant see all of my installed plugin, I only can view inactive plugin.
![enha](https://user-images.githubusercontent.com/756550/72439047-c1140600-37d8-11ea-83ba-8d53dbabe52f.jpg)

**Comment by bueltge** [comment]
@hilfans PLease try the plugin without all your other plugins and a default theme. After it works, add your theme and each plugin for plugin. So you should find the problem. The message (red) is not from this plugin and I haven't any idea where is the problem starts.

**Comment by hilfans** [comment]
hi, finally, the problem is coming from web server configuration and php module
thank you @bueltge  for the clue..

**Comment by GaryJones** [comment]
Was any further progress made on this issue? In my case, a client has a setup of 17 networks, and wants to find out which plugins and themes are unused in the install.

**Comment by bueltge** [comment]
Hi @GaryJones Currently no. I work only on the settings page, is done and need only here and there changes for code style. But, helping hands, PRs always greatly appreciated.
